reason-parser/src/reason_pprint_ast.ml

```diff
@@ -125,6 +125,7 @@ and ruleCategory =
      that it's easier just to always wrap them in parens.  *)
   | PotentiallyLowPrecedence of layoutNode
   | Simple of layoutNode
+  | InfixApplicationConcrete of ruleInfoData * string * ruleCategory * ruleCategory 
```

Comment:
We discussed in chat, that this would likely take the place of `SpecificInfixPrecedence`. Do you still believe this is the case? Is still seems there's a large overlap in their purposes.